src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -634,11 +634,13 @@ public void run()
          if (idleTimeout > 0L) {
             logPoolState("Before cleanup\t");
             final List<PoolBagEntry> bag = connectionBag.values(STATE_NOT_IN_USE);
-            int removable = bag.size() - config.getMinimumIdle();
-            for (PoolBagEntry bagEntry : bag) {
+            final int bagSize = bag.size();
+            int removable = bagSize - config.getMinimumIdle();
+            for (int i = 0; i < bagSize; i++) {
                if (removable <= 0) {
                   break;
                }
+               final PoolBagEntry bagEntry = bag.get(i);
```

Comment:
other side of coin here http://howtodoinjava.com/2013/03/26/performance-comparison-of-different-for-loops-in-java/#comment-866